helion/_compiler/compile_environment.py

```diff
@@ -272,6 +273,67 @@ def cached_create_unbacked_symint(
             self._symint_cache[key] = result
         return result
 
+    def get_tile_index_tensor_block_id(self, tensor: torch.Tensor) -> int | None:
+        """Return the originating ``tile.index`` block id if present."""
+        return tensor._tile_index_block_id  # type: ignore[attr-defined]
```

Comment:
Updated PR to patch the FakeTensor constructor to always set the `_tile_index_block_id` attribute (default to None), so that we can avoid the getattr here.
(an out-of-band dictionary mapping from FakeTensor to tile_index_block_id would be nicer I think, but FakeTensors currently don't have a stable ID to be used as the key 😞)